### DISCUSSION TITLE
PC1 always fails, reporting insufficient space on an empty 1TB nvme drive

### DISCUSSION THREAD

**Discussion by seccentral** [filecoin-project/lotus#5626] [Storage Provider]
I get out of space errors even if there should be no reason to:
My setup is as follows: 
I am on the latest lotus version (1.4.2) running on debian with this hardware config
80Gb ram (I know, its not 128 yet, will be when the other 64 kit arrives next week)
1TB pci3.0 nvme OS drive (this one holds the os, the lotus full chain, the proofs and the swap) 
1TB pci4.0 nvme seal drive (this one is completely dedicated to sealing)
A (for now) 5drive software md raid5 array of around 15TB 
Ryzen 2700X 
RTX 2080 
lotus compiled and installed as instructed in the docs. lotus daemon and miner running as user "fil". configs stored in default places ~/.lotus etc

Please bear in mind that initially, ie at the moment i issue the command `lotus-miner sectors pledge` the sealing drive is EMPTY. the output below is how it looks like after about 4hours when PC1 finishes and errors are thrown in the log
partitions setup and mount points
```
fil@filecoin:/filecoin/pcie40storage/lotus-miner/sealing$ df -h
Filesystem      Size  Used Avail Use% Mounted on
udev             40G     0   40G   0% /dev
tmpfs           7.9G   11M  7.9G   1% /run
/dev/nvme0n1p2  922G  768G  108G  88% /
tmpfs            40G     0   40G   0% /dev/shm
tmpfs           5.0M  4.0K  5.0M   1% /run/lock
tmpfs            40G     0   40G   0% /sys/fs/cgroup
/dev/nvme0n1p1  511M   13M  499M   3% /boot/efi
tmpfs           7.9G     0  7.9G   0% /run/user/0
/dev/md127       15T   86G   14T   1% /filecoin/hddSataStorage
/dev/nvme1n1p1  916G  673G  198G  78% /filecoin/pcie40storage
```

lotus storage setup 
```
fil@filecoin:/filecoin/pcie40storage/lotus-miner/sealing$ lotus-miner storage list
309523c5-e708-4d31-ba63-d1849f27e1de:
        [##                                                ] 830.5 GiB/14.44 TiB 5%
        Unsealed: 0; Sealed: 0; Caches: 0; Reserved: 0 B
        Weight: 10; Use: Store
        Local: /filecoin/hddSataStorage/storage/
        URL: http://127.0.0.1:2345/remote

f9b30223-d2d9-4231-9262-4c0e9289eed7:
        [#######################################           ] 718.7 GiB/915.9 GiB 78%
        Unsealed: 2; Sealed: 2; Caches: 2; Reserved: 0 B
        Weight: 10; Use: Seal 
        Local: /filecoin/pcie40storage/lotus-miner/sealing/
        URL: http://127.0.0.1:2345/remote
```

full sector log: 
```
fil@filecoin:/filecoin/pcie40storage/lotus-miner/sealing$ lotus-miner sectors status --log 2
SectorID:       2
Status:         PreCommit1
CIDcommD:       <nil>
CIDcommR:       <nil>
Ticket:         f28bf70f2b0e64c1c94d47a266bb60a1d44ff5071d8c7636880e1ce7964d2bee
TicketH:        511159
Seed:
SeedH:          0
Precommit:      <nil>
Commit:         <nil>
Proof:
Deals:          [0]
Retries:        0
--------
Event Log:
0.      2021-02-18 19:09:37 +0200 EET:  [event;sealing.SectorStartCC]   {"User":{"ID":2,"SectorType":8,"Pieces":[{"Piece":{"Size":34359738368,"PieceCID":{"/":"baga6ea4seaqao7s73y24kcutaosvacpdjgfe5pw76ooefnyqw4ynr3d2y6x2mpq"}},"DealInfo":null}]}}
1.      2021-02-18 19:09:37 +0200 EET:  [event;sealing.SectorPacked]    {"User":{"FillerPieces":null}}
2.      2021-02-18 19:09:37 +0200 EET:  [event;sealing.SectorTicket]    {"User":{"TicketValue":"8ov3DysOZMHJTUeiZrtgodRP9QcdjHY2iA4c55ZNK+4=","TicketEpoch":511159}}
3.      2021-02-18 19:49:28 +0200 EET:  [event;sealing.SectorRestart]   {"User":{}}
4.      2021-02-18 22:23:10 +0200 EET:  [event;sealing.SectorPreCommit1]        {"User":{"PreCommit1Out":"eyJyZWdpc3RlcmVkX3Byb29mIjoiU3RhY2tlZERyZzMyR2lCVjFfMSIsImxhYmVscyI6eyJTdGFja2VkRHJnMzJHaUJWMSI6eyJsYWJlbHMiOlt7InBhdGgiOiIvZmlsZWNvaW4vcGNpZTQwc3RvcmFnZS9sb3R1cy1taW5lci9zZWFsaW5nL2NhY2hlL3MtdDAyMTYwNjgtMiIsImlkIjoibGF5ZXItMSIsInNpemUiOjEwNzM3NDE4MjQsInJvd3NfdG9fZGlzY2FyZCI6N30seyJwYXRoIjoiL2ZpbGVjb2luL3BjaWU0MHN0b3JhZ2UvbG90dXMtbWluZXIvc2VhbGluZy9jYWNoZS9zLXQwMjE2MDY4LTIiLCJpZCI6ImxheWVyLTIiLCJzaXplIjoxMDczNzQxODI0LCJyb3dzX3RvX2Rpc2NhcmQiOjd9LHsicGF0aCI6Ii9maWxlY29pbi9wY2llNDBzdG9yYWdlL2xvdHVzLW1pbmVyL3NlYWxpbmcvY2FjaGUvcy10MDIxNjA2OC0yIiwiaWQiOiJsYXllci0zIiwic2l6ZSI6MTA3Mzc0MTgyNCwicm93c190b19kaXNjYXJkIjo3fSx7InBhdGgiOiIvZmlsZWNvaW4vcGNpZTQwc3RvcmFnZS9sb3R1cy1taW5lci9zZWFsaW5nL2NhY2hlL3MtdDAyMTYwNjgtMiIsImlkIjoibGF5ZXItNCIsInNpemUiOjEwNzM3NDE4MjQsInJvd3NfdG9fZGlzY2FyZCI6N30seyJwYXRoIjoiL2ZpbGVjb2luL3BjaWU0MHN0b3JhZ2UvbG90dXMtbWluZXIvc2VhbGluZy9jYWNoZS9zLXQwMjE2MDY4LTIiLCJpZCI6ImxheWVyLTUiLCJzaXplIjoxMDczNzQxODI0LCJyb3dzX3RvX2Rpc2NhcmQiOjd9LHsicGF0aCI6Ii9maWxlY29pbi9wY2llNDBzdG9yYWdlL2xvdHVzLW1pbmVyL3NlYWxpbmcvY2FjaGUvcy10MDIxNjA2OC0yIiwiaWQiOiJsYXllci02Iiwic2l6ZSI6MTA3Mzc0MTgyNCwicm93c190b19kaXNjYXJkIjo3fSx7InBhdGgiOiIvZmlsZWNvaW4vcGNpZTQwc3RvcmFnZS9sb3R1cy1taW5lci9zZWFsaW5nL2NhY2hlL3MtdDAyMTYwNjgtMiIsImlkIjoibGF5ZXItNyIsInNpemUiOjEwNzM3NDE4MjQsInJvd3NfdG9fZGlzY2FyZCI6N30seyJwYXRoIjoiL2ZpbGVjb2luL3BjaWU0MHN0b3JhZ2UvbG90dXMtbWluZXIvc2VhbGluZy9jYWNoZS9zLXQwMjE2MDY4LTIiLCJpZCI6ImxheWVyLTgiLCJzaXplIjoxMDczNzQxODI0LCJyb3dzX3RvX2Rpc2NhcmQiOjd9LHsicGF0aCI6Ii9maWxlY29pbi9wY2llNDBzdG9yYWdlL2xvdHVzLW1pbmVyL3NlYWxpbmcvY2FjaGUvcy10MDIxNjA2OC0yIiwiaWQiOiJsYXllci05Iiwic2l6ZSI6MTA3Mzc0MTgyNCwicm93c190b19kaXNjYXJkIjo3fSx7InBhdGgiOiIvZmlsZWNvaW4vcGNpZTQwc3RvcmFnZS9sb3R1cy1taW5lci9zZWFsaW5nL2NhY2hlL3MtdDAyMTYwNjgtMiIsImlkIjoibGF5ZXItMTAiLCJzaXplIjoxMDczNzQxODI0LCJyb3dzX3RvX2Rpc2NhcmQiOjd9LHsicGF0aCI6Ii9maWxlY29pbi9wY2llNDBzdG9yYWdlL2xvdHVzLW1pbmVyL3NlYWxpbmcvY2FjaGUvcy10MDIxNjA2OC0yIiwiaWQiOiJsYXllci0xMSIsInNpemUiOjEwNzM3NDE4MjQsInJvd3NfdG9fZGlzY2FyZCI6N31dLCJfaCI6bnVsbH19LCJjb25maWciOnsicGF0aCI6Ii9maWxlY29pbi9wY2llNDBzdG9yYWdlL2xvdHVzLW1pbmVyL3NlYWxpbmcvY2FjaGUvcy10MDIxNjA2OC0yIiwiaWQiOiJ0cmVlLWQiLCJzaXplIjoyMTQ3NDgzNjQ3LCJyb3dzX3RvX2Rpc2NhcmQiOjd9LCJjb21tX2QiOls3LDEyNiw5NSwyMjIsNTMsMTk3LDEwLDE0NywzLDE2NSw4MCw5LDIyNyw3MywxMzgsNzgsMTkwLDIyMywyNDMsMTU2LDY2LDE4MywxNiwxODMsNDgsMjE2LDIzNiwxMjIsMTk5LDE3NSwxNjYsNjJdfQ=="}}
5.      2021-02-18 22:23:11 +0200 EET:  [event;sealing.SectorSealPreCommit2Failed]      {"User":{}}
        seal pre commit(2) failed: storage call error 0: presealing sector 2 (): Rust panic: no unwind information
6.      2021-02-18 22:24:11 +0200 EET:  [event;sealing.SectorRetrySealPreCommit2]       {"User":{}}
7.      2021-02-18 22:24:11 +0200 EET:  [event;sealing.SectorSealPreCommit2Failed]      {"User":{}}
        seal pre commit(2) failed: storage call error 0: presealing sector 2 (): Rust panic: no unwind information
8.      2021-02-18 22:25:11 +0200 EET:  [event;sealing.SectorRetrySealPreCommit2]       {"User":{}}
9.      2021-02-18 22:25:11 +0200 EET:  [event;sealing.SectorSealPreCommit2Failed]      {"User":{}}
        seal pre commit(2) failed: storage call error 0: presealing sector 2 (): Rust panic: no unwind information
10.     2021-02-18 22:26:11 +0200 EET:  [event;sealing.SectorRetrySealPreCommit2]       {"User":{}}
11.     2021-02-18 22:26:11 +0200 EET:  [event;sealing.SectorSealPreCommit2Failed]      {"User":{}}
        seal pre commit(2) failed: storage call error 0: presealing sector 2 (): Rust panic: no unwind information
12.     2021-02-18 22:27:11 +0200 EET:  [event;sealing.SectorRetrySealPreCommit1]       {"User":{}}
```

lotus-miner log 
```
{"level":"debug","ts":"2021-02-18T22:26:11.006+0200","logger":"stores","caller":"stores/index.go:396","msg":"not allocating on f9b30223-d2d9-4231-9262-4c0e9289eed7, out of space (available: 211768414208, need: 518832049356)"}
{"level":"error","ts":"2021-02-18T22:26:11.006+0200","logger":"advmgr","caller":"sector-storage/sched.go:404","msg":"trySched(1) req.sel.Ok error: finding best alloc storage:\n    github.com/filecoin-project/lotus/extern/sector-storage.(*allocSelector).Ok\n        /home/fil/lotus/extern/sector-storage/selector_alloc.go:55\n  - no good path found:\n    github.com/filecoin-project/lotus/extern/sector-storage/stores.(*Index).StorageBestAlloc\n        /home/fil/lotus/extern/sector-storage/stores/index.go:414"}
{"level":"debug","ts":"2021-02-18T22:26:11.006+0200","logger":"stores","caller":"stores/index.go:396","msg":"not allocating on f9b30223-d2d9-4231-9262-4c0e9289eed7, out of space (available: 211768414208, need: 518832049356)"}
{"level":"error","ts":"2021-02-18T22:26:11.006+0200","logger":"advmgr","caller":"sector-storage/sched.go:404","msg":"trySched(1) req.sel.Ok error: finding best alloc storage:\n    github.com/filecoin-project/lotus/extern/sector-storage.(*allocSelector).Ok\n        /home/fil/lotus/extern/sector-storage/selector_alloc.go:55\n  - no good path found:\n    github.com/filecoin-project/lotus/extern/sector-storage/stores.(*Index).StorageBestAlloc\n        /home/fil/lotus/extern/sector-storage/stores/index.go:414"}
{"level":"debug","ts":"2021-02-18T22:26:11.006+0200","logger":"advmgr","caller":"sector-storage/sched.go:449","msg":"SCHED windows: [{allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]}]"}
{"level":"debug","ts":"2021-02-18T22:26:11.006+0200","logger":"advmgr","caller":"sector-storage/sched.go:450","msg":"SCHED Acceptable win: [[]]"}
{"level":"debug","ts":"2021-02-18T22:26:11.043+0200","logger":"advmgr","caller":"sector-storage/worker_local.go:129","msg":"acquired sector {{216068 2} 8} (e:6; a:0): {{0 0}  /filecoin/pcie40storage/lotus-miner/sealing/sealed/s-t0216068-2 /filecoin/pcie40storage/lotus-miner/sealing/cache/s-t0216068-2}"}
thread '<unnamed>' panicked at 'failed to create ColumnTreeBuilder: ClError(BusIdNotAvailable)', /home/fil/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-5.4.0/src/stacked/vanilla/proof.rs:516:23
thread '<unnamed>' panicked at 'failed to send columns: "SendError(..)"', /home/fil/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-5.4.0/src/stacked/vanilla/proof.rs:502:34
{"level":"warn","ts":"2021-02-18T22:26:11.450+0200","logger":"sectors","caller":"storage-sealing/fsm.go:535","msg":"sector 2 got error event sealing.SectorSealPreCommit2Failed: seal pre commit(2) failed: storage call error 0: presealing sector 2 (): Rust panic: no unwind information"}
{"level":"info","ts":"2021-02-18T22:26:11.487+0200","logger":"sectors","caller":"storage-sealing/states_failed.go:25","msg":"SealPreCommit2Failed(2), waiting 59.512585706s before retrying"}
{"level":"debug","ts":"2021-02-18T22:27:11.038+0200","logger":"advmgr","caller":"sector-storage/sched.go:354","msg":"SCHED 2 queued; 2 open windows"}
{"level":"debug","ts":"2021-02-18T22:27:11.038+0200","logger":"stores","caller":"stores/index.go:396","msg":"not allocating on f9b30223-d2d9-4231-9262-4c0e9289eed7, out of space (available: 211768414208, need: 518832049356)"}
{"level":"error","ts":"2021-02-18T22:27:11.038+0200","logger":"advmgr","caller":"sector-storage/sched.go:404","msg":"trySched(1) req.sel.Ok error: finding best alloc storage:\n    github.com/filecoin-project/lotus/extern/sector-storage.(*allocSelector).Ok\n        /home/fil/lotus/extern/sector-storage/selector_alloc.go:55\n  - no good path found:\n    github.com/filecoin-project/lotus/extern/sector-storage/stores.(*Index).StorageBestAlloc\n        /home/fil/lotus/extern/sector-storage/stores/index.go:414"}
{"level":"debug","ts":"2021-02-18T22:27:11.038+0200","logger":"stores","caller":"stores/index.go:396","msg":"not allocating on f9b30223-d2d9-4231-9262-4c0e9289eed7, out of space (available: 211768414208, need: 518832049356)"}
{"level":"error","ts":"2021-02-18T22:27:11.038+0200","logger":"advmgr","caller":"sector-storage/sched.go:404","msg":"trySched(1) req.sel.Ok error: finding best alloc storage:\n    github.com/filecoin-project/lotus/extern/sector-storage.(*allocSelector).Ok\n        /home/fil/lotus/extern/sector-storage/selector_alloc.go:55\n  - no good path found:\n    github.com/filecoin-project/lotus/extern/sector-storage/stores.(*Index).StorageBestAlloc\n        /home/fil/lotus/extern/sector-storage/stores/index.go:414"}
{"level":"debug","ts":"2021-02-18T22:27:11.038+0200","logger":"stores","caller":"stores/index.go:396","msg":"not allocating on f9b30223-d2d9-4231-9262-4c0e9289eed7, out of space (available: 211768414208, need: 518832049356)"}
{"level":"error","ts":"2021-02-18T22:27:11.038+0200","logger":"advmgr","caller":"sector-storage/sched.go:404","msg":"trySched(1) req.sel.Ok error: finding best alloc storage:\n    github.com/filecoin-project/lotus/extern/sector-storage.(*allocSelector).Ok\n        /home/fil/lotus/extern/sector-storage/selector_alloc.go:55\n  - no good path found:\n    github.com/filecoin-project/lotus/extern/sector-storage/stores.(*Index).StorageBestAlloc\n        /home/fil/lotus/extern/sector-storage/stores/index.go:414"}
{"level":"debug","ts":"2021-02-18T22:27:11.038+0200","logger":"stores","caller":"stores/index.go:396","msg":"not allocating on f9b30223-d2d9-4231-9262-4c0e9289eed7, out of space (available: 211768414208, need: 518832049356)"}
{"level":"error","ts":"2021-02-18T22:27:11.038+0200","logger":"advmgr","caller":"sector-storage/sched.go:404","msg":"trySched(1) req.sel.Ok error: finding best alloc storage:\n    github.com/filecoin-project/lotus/extern/sector-storage.(*allocSelector).Ok\n        /home/fil/lotus/extern/sector-storage/selector_alloc.go:55\n  - no good path found:\n    github.com/filecoin-project/lotus/extern/sector-storage/stores.(*Index).StorageBestAlloc\n        /home/fil/lotus/extern/sector-storage/stores/index.go:414"}
{"level":"debug","ts":"2021-02-18T22:27:11.038+0200","logger":"advmgr","caller":"sector-storage/sched.go:449","msg":"SCHED windows: [{allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]}]"}
{"level":"debug","ts":"2021-02-18T22:27:11.038+0200","logger":"advmgr","caller":"sector-storage/sched.go:450","msg":"SCHED Acceptable win: [[] []]"}
```

Please help me understand what am i doing wrong and how much space sealing one sector actually needs, thank you

**Comment by stuberman** [comment]
It seems like your two sectors are stuck before they can complete.

>         Unsealed: 2; Sealed: 2; Caches: 2; Reserved: 0 B
>         Weight: 10; Use: Seal 
>         Local: /filecoin/pcie40storage/lotus-miner/sealing/

Based on the sector log it looks like PC2 failed and left you two stuck sectors. The rust panic is troubling.
It is OK to get an out of space error if the miner is trying to instantiate a second task when there is not enough resource to run a new task
This is the error I would focus on "Rust panic: no unwind information"
Are you using a GPU? If so, what type.

**Reply by stuberman** [reply]
I suggest you take the error in quotes and search in our Slack site. Lots of advice and tips there.

**Reply by seccentral** [reply]
the gpu is an Nvidia rtx 2080 that (i think) i installed properly: 
```
root@filecoin:/filecoin/pcie40storage/lotus-miner/sealing/cache# lsmod | grep nvid
nvidia_drm             49152  0
nvidia_modeset       1097728  2 nvidia_drm
nvidia              18653184  18 nvidia_modeset
drm_kms_helper        208896  1 nvidia_drm
drm                   495616  3 drm_kms_helper,nvidia_drm
ipmi_msghandler        65536  2 ipmi_devintf,nvidia
```
```
root@filecoin:~# dpkg -l nvidia-driver
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
ii  nvidia-driver  418.181.07-1 amd64        NVIDIA metapackage
```
```
root@filecoin:~# modinfo nvidia
filename:       /lib/modules/4.19.0-14-amd64/updates/dkms/nvidia.ko
alias:          char-major-195-*
version:        460.39
supported:      external
license:        NVIDIA
srcversion:     6D32E1B1DFCA8DA07AD1677
alias:          pci:v000010DEd*sv*sd*bc03sc02i00*
alias:          pci:v000010DEd*sv*sd*bc03sc00i00*
depends:        
retpoline:      Y
name:           nvidia
vermagic:       4.19.0-14-amd64 SMP mod_unload modversions 
parm:           NvSwitchRegDwords:NvSwitch regkey (charp)
parm:           NvSwitchBlacklist:NvSwitchBlacklist=uuid[,uuid...] (charp)
parm:           nv_cap_enable_devfs:Enable (1) or disable (0) nv-caps devfs support. Default: 1 (int)
parm:           NVreg_ResmanDebugLevel:int
parm:           NVreg_RmLogonRC:int
parm:           NVreg_ModifyDeviceFiles:int
parm:           NVreg_DeviceFileUID:int
parm:           NVreg_DeviceFileGID:int
parm:           NVreg_DeviceFileMode:int
parm:           NVreg_InitializeSystemMemoryAllocations:int
parm:           NVreg_UsePageAttributeTable:int
parm:           NVreg_RegisterForACPIEvents:int
parm:           NVreg_EnablePCIeGen3:int
parm:           NVreg_EnableMSI:int
parm:           NVreg_TCEBypassMode:int
parm:           NVreg_EnableStreamMemOPs:int
parm:           NVreg_EnableBacklightHandler:int
parm:           NVreg_RestrictProfilingToAdminUsers:int
parm:           NVreg_PreserveVideoMemoryAllocations:int
parm:           NVreg_EnableS0ixPowerManagement:int
parm:           NVreg_S0ixPowerManagementVideoMemoryThreshold:int
parm:           NVreg_DynamicPowerManagement:int
parm:           NVreg_DynamicPowerManagementVideoMemoryThreshold:int
parm:           NVreg_EnableUserNUMAManagement:int
parm:           NVreg_MemoryPoolSize:int
parm:           NVreg_KMallocHeapMaxSize:int
parm:           NVreg_VMallocHeapMaxSize:int
parm:           NVreg_IgnoreMMIOCheck:int
parm:           NVreg_NvLinkDisable:int
parm:           NVreg_EnablePCIERelaxedOrderingMode:int
parm:           NVreg_RegisterPCIDriver:int
parm:           NVreg_RegistryDwords:charp
parm:           NVreg_RegistryDwordsPerDevice:charp
parm:           NVreg_RmMsg:charp
parm:           NVreg_GpuBlacklist:charp
parm:           NVreg_TemporaryFilePath:charp
```
```
root@filecoin:~# nvidia-detect 
Detected NVIDIA GPUs:
0e:00.0 VGA compatible controller [0300]: NVIDIA Corporation TU104 [GeForce RTX 2080 Rev. A] [10de:1e87] (rev a1)

Checking card:  NVIDIA Corporation TU104 [GeForce RTX 2080 Rev. A] (rev a1)
Your card is supported by the default drivers.
It is recommended to install the
    nvidia-driver
package.
```

Regarding the sectors: I know it should be ok to fail when running a second one and not having enough space, but this happens even if I run only one at a time. I just didn't feel like erasing everything and starting from the beginning when I posted the logs, take my word for it ? 😇

**Reply by stuberman** [reply]
Have you run `nvidia-smi` to see status? Looks like you might be running the default [Nouveau driver](https://linuxconfig.org/how-to-disable-nouveau-nvidia-driver-on-ubuntu-18-04-bionic-beaver-linux)

**Reply by seccentral** [reply]
The problem was super easy to fix, but the fix was found by chance. spend a few days searching for this as nobody seems to have noticed to inlcude installing the "nvidia-opencl-icd" package as a mandatory prerequisite for all the gpu compute required stages. 
  
**### PLEASE update your docs.**   

Maybe everybody else was developing on anything BUT debian, but on debian, this is not by default included when installing the driver metapackage, nor is it included as a dependency of any other package that you list in your documentation .
https://docs.filecoin.io/get-started/lotus/installation/#software-dependencies
The docs state mesa. Mesa is not enough, apparently.

```
TL/DR: FIX 
user@host$: apt install nvidia-opencl-icd
```

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
what does clinfo spit out?
